src/wallet/wallet.cpp

```diff
@@ -1318,6 +1318,31 @@ CAmount CWallet::GetDebit(const CTxIn &txin, const isminefilter& filter) const
     return 0;
 }
 
+std::optional<CNameScript> CWallet::GetNameDebit(const CTxIn &txin, const isminefilter& filter) const
+{
+    {
```

Comment:
What is the reason for this extra block here, which spans the entire function anyway?  I think you should just remove it (perhaps it is left over from some refactorings?).

Reply:
I have no clue what that block is intended to do, but it's copied from `GetDebit()` where it's also present (and I also don't know why it's there either).  Maybe it's related to locking?  I left it intact because doing so makes it easier to audit that the structure matches upstream `GetDebit()`, and I assume upstream may know something I don't.

Reply:
Yes, it is about locking.  But since you hold the lock the entire function anyway, there's no point in it.  Most likely this is a left over from previous refactorings.  If you explicitly want to keep the structure the same, that's fine with me (although it is pretty straight-forward that removing the block changes nothing).

Reply:
Yes, I prefer to keep the structure the same.  (Though I guess feel free to submit a patch upstream for `GetDebit()`.)

Reply:
Then it is fine with me, full ACK.